### PR TITLE
PR: Implement support for Rayleigh-Jeans spectral distribution. 

### PR DESCRIPTION
**Pull request by Alt-Shivam**
(cherry picked from commit 285fef1b3cbab66064244ef92259de7bf5122783)
Related to issue #919

### CODE REVIEW

**Comment by Alt-Shivam** [comment]
@KelSolaar sir 
I hope I am going right.
Please have a look and suggest changes, If any

**Comment by Alt-Shivam** [comment]
@KelSolaar sir 
Please trigger workflows once again.

**Comment by Alt-Shivam** [comment]
Resolved all the errors 
except this 
image


because there is no trailing whitespace at that location.

**Comment by KelSolaar** [comment]
Hi @Alt-Shivam,

The white space is here: `where "c" is the speed of light; `

Even with that fixed, the docstring is incorrect, it should have proper parameters described.

There is also no reason to have a new module header middle of the module, i.e. this should not be here:

```python
"""
Rayleigh-Jeans spectral distribution
==============================

Defines the objects to compute the Rayleigh–Jeans law approximation to
the spectral radiance of electromagnetic radiation as a function of
wavelength from a black body at a given temperature through
classical arguments.
"""
```

Before I start to review, would it be please possible to add:

- Unit tests in *colour/colorimetry/tests/test_blackbody.py*
- Examples in *colour/examples/colorimetry/examples_blackbody.py*

Cheers,

Thomas

File: colour/colorimetry/blackbody.py [hunk 1]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
         wavelengths,
         name=f"{temperature}K Blackbody",
     )
+
+
+def rayleigh_jeans_law(
+    wavelength,
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:606]: We will need the typing annotations like for the `planck_law` definition.

File: colour/colorimetry/blackbody.py [hunk 2]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+    temperature,
+) -> FloatingOrNDArray:
+    """
+    Returns the spectral radiance of electromagnetic radiation as a
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:610]: We are now using imperative docstring style, thus `Returns the ...` should be `Return the ...`

File: colour/colorimetry/blackbody.py [hunk 3]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+    function of wavelength from a black body at a given temperature
+    :math:`T[K]` in a medium having index of refraction :math:`n`.
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:612]: ``index of refraction :math:`n`.`` --> ``index of refraction :math:`n` according to *Rayleigh–Jeans* law.``

File: colour/colorimetry/blackbody.py [hunk 4]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+    :class:`numpy.floating` or :class:`numpy.ndarray`
+        Radiance in *watts per steradian per square metre* (:math:`W/sr/m^2`).
+
+    Notes
+    -----
+    -   The following form implementation is expressed in term of
+        wavelength.
+    -   The SI unit of radiance is *watts per steradian per square metre*
+        (:math:`W/sr/m^2`).
+
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:632]: We will need a `References` section here. You can use ``:cite:`Wikipedia2003f`` and the following top level reference:

```reStructuredText
-   :cite:`Wikipedia2003f` : Wikipedia. (2003). Rayleigh–Jeans law. Retrieved
    February 12, 2022, from https://en.wikipedia.org/wiki/Rayleigh–Jeans_law
```

File: colour/colorimetry/blackbody.py [hunk 5]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+    c = CONSTANT_LIGHT_SPEED
+    k_B = CONSTANT_BOLTZMANN
+
+    B = (2 * c * k_B * T) / (λ**4)
+
+    return B
+
+
+def sd_rayleigh_jeans(
+    temperature,
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:651]: Ditto for the typing annotations! :)

File: colour/colorimetry/blackbody.py [hunk 6]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+    shape=SPECTRAL_SHAPE_DEFAULT,
+) -> SpectralDistribution:
+    """
+    Returns the spectral distribution of the Spectral radiance for
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:655]: Same for `Returns` vs `Return` and also `Spectral` is capitalised.

File: colour/colorimetry/blackbody.py [hunk 7]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+    given temperature :math:`T[K]` with values in
+    *watts per steradian per square metre per nanometer*
+    (:math:`W/sr/m^2/nm`).
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:658]: ` according to *Rayleigh–Jeans* law.`

File: colour/colorimetry/blackbody.py [hunk 8]
```diff
@@ -594,3 +600,513 @@ def sd_blackbody(
...
+        Spectral shape used to create the spectral distribution of the
+        planckian radiator.
+
+    Returns
+    -------
+    SpectralDistribution
+        Rayleigh-Jeans spectral distribution with values in
+        *watts per steradian per square metre per nanometer*
+        (:math:`W/sr/m^2/nm`).
+
```
Comment (KelSolaar) [inline colour/colorimetry/blackbody.py:674]: We can copy the `References` section from the function above.

File: docs/colour.colorimetry.rst [hunk 9]
```diff
@@ -53,6 +53,7 @@ Spectral Data Generation
     sd_blackbody
     sd_constant
     sd_ones
+    sd_rayleigh_jeans
```
Comment (KelSolaar) [inline docs/colour.colorimetry.rst:56]: the `rayleigh_jeans_law` definition should be documented in the section just after.

**Review by KelSolaar — CHANGES_REQUESTED** [review]
Hi @Alt-Shivam,

Thanks for the PR, I left a few minor stylistic comments. It would be interesting
to put an example in `colour/examples/colorimetry/examples_blackbody.py` maybe!

Cheers,

Thomas

**Comment by Alt-Shivam** [comment]
Unit tests in colour/colorimetry/tests/test_blackbody.py
Examples in colour/examples/colorimetry/examples_blackbody.py
I am still working on them.

And thanks for the constant support sir 😊

**Comment by Alt-Shivam** [comment]
Hey @KelSolaar sir 
Added Unit test for Rayleigh-Jeans law 
but still struggling with the test data.
Constantly trying for the last 3days.
But still........

Can you please help me out. 
Thanks 😊

**Comment by KelSolaar** [comment]
Can you please do not merge develop in your branch please? Always rebase! Is it possible to reset hard your branch to before the merge you did and force-push to pop the merge out?

**Comment by Alt-Shivam** [comment]
Again same mistake 😶
Merged develop while resolving conflicts. 
Will try to figure out what can be done now .
Thanks sir

**Comment by KelSolaar** [comment]
Closing in favour of #995 as this PR is unfortunately not in a state where it can be merged.

**Pull request closed without merging**